### PR TITLE
SPT-4287 Write to read only fields

### PR DESCRIPTION
**Pull request by mbutlerswim**
Enables the ability to write to read only fields via a parameter of the driver.  Also added in better handling for json fields.

Updated documentation as well.

### CODE REVIEW

File: swimlane/core/fields/base/field.py [hunk 1]
```diff
@@ -100,7 +100,7 @@ def cast_to_bulk_modify(self, value):
 
     def validate_value(self, value):
         """Validate value is an acceptable type during set_python operation"""
-        if self.readonly:
+        if self.readonly and not self._swimlane._write_to_read_only:
```
Comment (dmikov) [inline swimlane/core/fields/base/field.py]: I do not see where this value is set.

Comment (mbutlerswim) [reply swimlane/core/fields/base/field.py]: It is set when the Swimlane client is initialized here: https://github.com/mbutlerswim/swimlane-python/blob/write-read-only/swimlane/core/client.py#L108

Comment (dmikov) [reply swimlane/core/fields/base/field.py]: But it is not a client nor does it inherit from it, so self will not be populated for field class

Comment (mbutlerswim) [reply swimlane/core/fields/base/field.py]: The Field class inherits the SwimlaneResolver class which applies the swimlane client in the property self._swimlane.  So the _write_to_read_only property is available to the Field class at self._swimlane._write_to_read_only.

Comment (mbutlerswim) [reply swimlane/core/fields/base/field.py]: I have also added a test now as well.

**Pull request merged**